### PR TITLE
feat(function): add json() method to HTTP trigger Response class

### PR DESCRIPTION
**Pull request by htuna07**
Add res.json(body) method to the function HTTP trigger Response class, mirroring Express-style API. Unlike send(), json() always serializes the body with JSON.stringify and sets Content-type: application/json, regardless of the body type (object, array, primitive, boolean, null).

Usage: res.status(200).json({ ok: true })

Add tests covering objects, arrays, numbers, booleans, null, and status code chaining via status().

### CODE REVIEW

File: packages/api/function/queue/node/src/http.ts [hunk 1]
```diff
@@ -133,6 +133,15 @@ export class Response {
...
   ) {}
 
+  async json(body: any): Promise<void> {
+    const chunk = Buffer.from(JSON.stringify(body));
+    await this.writeHead(this.statusCode || 200, this.statusMessage || "OK", {
+      "Content-type": "application/json",
+      "Content-length": String(Buffer.byteLength(chunk))
+    });
+    return this.end(chunk, "utf-8");
+  }
```
[the remark this answers is not in the document]
Comment (htuna07) [reply packages/api/function/queue/node/src/http.ts:143]: @copilot apply changes based on [this feedback](https://github.com/spica-engine/spica/pull/1779#discussion_r3109460968)

**Review by idriska — APPROVED** [review]
